backend/src/main/java/de/bund/digitalservice/ris/norms/application/port/input/LoadArticleHtmlUseCase.java

```diff
@@ -1,12 +1,21 @@
 package de.bund.digitalservice.ris.norms.application.port.input;
 
+import de.bund.digitalservice.ris.norms.application.exception.ArticleNotFoundException;
+import de.bund.digitalservice.ris.norms.application.exception.NormNotFoundException;
 import java.time.Instant;
-import java.util.Optional;
 import org.springframework.lang.Nullable;
 
 /** Use case for loading a norm's article */
 public interface LoadArticleHtmlUseCase {
-  Optional<String> loadArticleHtml(Query query);
+  /**
+   * Loads the HTML of an article from a norm.
+   *
+   * @param query The article to load
+   * @return HTML string of the article
+   * @throws NormNotFoundException If the norm is not found
+   * @throws ArticleNotFoundException If the norm exists but does not contain the article
+   */
+  String loadArticleHtml(Query query) throws ArticleNotFoundException, NormNotFoundException;
```

Comment:
@VictorDelCampo I noticed that not all of our use cases declare that they're throwing exceptions. I added the `throws` here because I figured it would be useful to know how loading the article HTML can go wrong but not sure what the typical approach here is.

Reply:
In java you have two types of exceptions, 

1. checked exceptions that represent errors outside the control of the program and that a well-written application should anticipate and recover from (like for example file not found `FileNotFoundException`), meaning if you write code using a method from a library that may throw that checked exception you should handle that case in your code. These exceptions are checked at compile-time, meaning if you don't handle them you will get an error.
2. runtime exceptions that are used to indicate programming errors (like the famous `NullPointerException` or `ArrayIndexOutOfBoundsException`). Java does not verify these unchecked exceptions at compile-time, meaning you don't need to handle them and therefore are not needed in the `throws` declaration of a method.


The general praxis is that runtime exceptions (the ones we have) are not declared in the `throws` definition but can be added, specially for readability of the code and giving the caller some extra information.

It should be IMO a team decision and probably a topic for the dev weekly?

Reply:
Thanks for the detailed explanation!